pkg/kv/kvserver/rafttrace/rafttrace.go

```diff
@@ -88,17 +90,32 @@ type RaftTracer struct {
 	m             syncutil.Map[kvpb.RaftIndex, traceValue]
 	numRegistered atomic.Int64
 	ctx           context.Context
+	st            *cluster.Settings
 }
 
 // NewRaftTracer creates a new RaftTracer with the given context.
-func NewRaftTracer(ctx context.Context) *RaftTracer {
-	return &RaftTracer{ctx: ctx}
+func NewRaftTracer(ctx context.Context, st *cluster.Settings) *RaftTracer {
+	return &RaftTracer{ctx: ctx, st: st}
 }
 
+// MaxConcurrentRaftTraces is the maximum number of traces that can be active at
+// any time. Additional traces will be ignored until the number of traces drops.
+// Having too many active traces will negatively impact performance as we
+// iterate over all of them for each message. 10 is a reasonable default that
+// balances usefulness with performance impact. It isn't expected that this
+// limit will normally be hit.
+var MaxConcurrentRaftTraces = settings.RegisterIntSetting(
```

Comment:
Can we have some escape hatch to remove all the actively traced things, e.g. when this setting goes to 0?

I'm not sure how to do this exactly. Maybe in places where we check this setting, we could have a special case: if the max drops to 0, clear the data structure.

Reply:
I renamed Close to FlushAll to clean up all the traces. Your concern from above about the race is still valid, and I'm not 100% sure how to deal with it, but I'm